azext_iot/iothub/providers/state.py

```diff
@@ -0,0 +1,273 @@
+
+        identities = iot_device_list(cmd=self.cmd, hub_name=self.orig_hub, resource_group_name=self.orig_rg, auth_type_dataplane=self.auth_type)
+
+        configs = iot_hub_configuration_list(cmd=self.cmd, hub_name=self.orig_hub, resource_group_name=self.orig_rg, auth_type_dataplane=self.auth_type)
+        for c in configs:
+            iot_hub_configuration_create(cmd=self.cmd, config_id=c["id"], content=json.dumps(c["content"]), hub_name=self.hub_name, target_condition=c["targetCondition"], \
+                priority=c["priority"], labels=json.dumps(c["labels"]), metrics=json.dumps(c["metrics"]), resource_group_name=self.rg, auth_type_dataplane=self.auth_type)
+
+        for id in identities: 
+
+            # primary and secondary keys show up in the "show" output but not in the "list" output
+            if id["authenticationType"] == "sas":
+                id2 = iot_device_show(self.cmd, id["deviceId"], self.orig_hub, self.orig_rg, None, self.auth_type)
+                id["symmetricKey"] = id2["authentication"]["symmetricKey"]
+
+            twin = iot_device_twin_show(self.cmd, id["deviceId"], self.orig_hub, self.orig_rg, None, self.auth_type)
+
+            self.upload_device_identity(id)
+            iot_device_twin_replace(self.cmd, id["deviceId"], json.dumps(twin), self.hub_name)
+
+            module_objs = iot_device_module_list(cmd=self.cmd, device_id=id["deviceId"], hub_name=self.orig_hub, resource_group_name=self.orig_rg, auth_type_dataplane=self.auth_type)
+
+            for module in module_objs:
+                module = vars(module)
+
+                if(module["module_id"] not in ["$edgeAgent", "$edgeHub"]):
+
+                    module_twin = iot_device_module_twin_show(cmd=self.cmd, device_id=id["deviceId"], module_id=module["module_id"], hub_name=self.orig_hub, \
+                        resource_group_name=self.orig_rg, auth_type_dataplane=self.auth_type)
+
+                    module["authenticationType"] = module_twin["authenticationType"]
+                    if (module["authenticationType"] == "sas"):
+                        module2 = iot_device_module_show(cmd=self.cmd, device_id=id["deviceId"], module_id=module["module_id"], hub_name=self.orig_hub, \
+                            resource_group_name=self.orig_rg, auth_type_dataplane=self.auth_type)
+                        module["symmetricKey"] = module2["authentication"]["symmetricKey"]
+
+                    if (module["authenticationType"] == "certificateAuthority"):
+                        module2 = iot_device_module_show(cmd=self.cmd, device_id=id["deviceId"], module_id=module["module_id"], hub_name=self.orig_hub, \
+                            resource_group_name=self.orig_rg, auth_type_dataplane=self.auth_type)
+                        module["x509Thumbprint"] = module2["authentication"]["x509Thumbprint"]
+
+                    print("uploading module '{0}' in device '{1}'.".format(module["module_id"], id["deviceId"]))
```

Comment:
In most cases you do not want to use print. Instead use the logger methods like logger.info(), logger.warning() etc